### PR TITLE
[14/N] Fix clang-tidy warnings in jit

### PR DESCRIPTION
**Pull request by cyyever**
Follows  #132411

cc @EikanWang @jgong5

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/operators/misc.cpp [hunk 1]
```diff
@@ -3,9 +3,9 @@
 #include <torch/csrc/jit/tensorexpr/operators/misc.h>
 #include <torch/csrc/jit/tensorexpr/tensor.h>
 
-namespace torch {
-namespace jit {
-namespace tensorexpr {
+#include <utility>
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/operators/misc.cpp:6]: Did you intend to include the utility IN a namespace?

File: torch/csrc/jit/tensorexpr/operators/misc.cpp [hunk 2]
```diff
@@ -160,7 +159,7 @@ ExprHandle demoteOutput(
   return e;
 }
 
-std::optional<TensorInfo> getTensorInfo(BufHandle b) {
+std::optional<TensorInfo> getTensorInfo(const BufHandle& b) {
   std::vector<int64_t> dims;
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/operators/misc.cpp:163]: Add a reserve here while you are editing this file

File: torch/csrc/jit/tensorexpr/operators/misc.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/operators/misc.cpp:6]: It is above the concatenated namespace

Comment (Skylion007) [reply torch/csrc/jit/tensorexpr/operators/misc.cpp:6]: but below tensorexpr

Comment (cyyever) [reply torch/csrc/jit/tensorexpr/operators/misc.cpp]: Fixed

File: torch/csrc/jit/tensorexpr/operators/misc.cpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/operators/misc.cpp:161]: Added

File: torch/csrc/jit/tensorexpr/operators/misc.cpp [hunk 3]
```diff
@@ -271,7 +270,7 @@ ExprHandle scalarOrConstant(const ArgValue& v) {
   return constant(v);
 }
 
-ExprHandle broadcast(BufHandle b, const std::vector<ExprHandle>& axes) {
+ExprHandle broadcast(const BufHandle& b, const std::vector<ExprHandle>& axes) {
   return b.load(computeIndicesToBroadcast(axes, b.dims()));
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/operators/misc.cpp:274]: b.load is a const method?

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**